src/sonic-yang-mgmt/sonic_yang_ext.py

```diff
@@ -160,6 +164,103 @@ def _preProcessYang(self, moduleName, module):
             raise e
         return
 
+    def _compileUsesClauseRefineApply(self, refine, model):
+        for item in model:
+            if item["@name"] == refine["@target-node"]:
+                # NOTE: We only handle 'description' and 'mandatory'
+                if refine.get('description') is not None:
+                    item["description"] = refine["description"]
+                if refine.get('mandatory') is not None:
+                    item["mandatory"] = refine["mandatory"]
+                return
+
+    def _compileUsesClauseRefine(self, refine, model):
+        # Nothing to refine
+        if refine is None:
+            return
+
+        # Convert to list if not already
+        if not isinstance(refine, list):
```

Comment:
>list [](http://example.com/codeflow?start=34&length=4)

Instead of checking not a list, can we check if one of allowed type options?

Reply:
The point of this is to treat everything as a list later in the code by converting it into a single-element list.  This is just a way to simplify the code below.  Its not meant as a sanity check.  It would be less obvious the intent to make your suggested change.